serf/serf.go

```diff
@@ -781,15 +781,33 @@ func (s *Serf) Members() []Member {
 	return members
 }
 
-// RemoveFailedNode forcibly removes a failed node from the cluster
+//RemoveFailedNode is a backwards compatabile form
+// of ForceLeave
+func (s *Serf) RemoveFailedNode(node string) error {
+	if err := s.ForceLeave(node, false); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *Serf) RemoveFailedNodePrune(node string) error {
+	if err := s.ForceLeave(node, true); err != nil {
+		return err
+	}
+	return nil
+
+}
+
+// ForceLeave forcibly removes a failed node from the cluster
 // immediately, instead of waiting for the reaper to eventually reclaim it.
 // This also has the effect that Serf will no longer attempt to reconnect
 // to this node.
-func (s *Serf) RemoveFailedNode(node string) error {
+func (s *Serf) ForceLeave(node string, prune bool) error {
```

Comment:
Do you need to make this method exported, or can it be named `forceLeave()` and left unexported?

Reply:
I can make it unexported!